code/game/objects/structures/mannequin.dm

```diff
@@ -0,0 +1,1298 @@
+		obj_overlays["[MANNEQUIN_HAND_LAYER]-[index]"] = O
+	else
+		overlays.Remove(O)
+		O.overlays.len = 0
+
+	var/obj/item/I = get_held_item_by_index(index)
+
+	if(I)
+		var/t_state = I.item_state
+		var/t_inhand_state = I.inhand_states[get_direction_by_index(index)]
+		var/icon/check_dimensions = new(t_inhand_state)
+		if(!t_state)
+			t_state = I.icon_state
+
+		O.name = "[index]"
+		O.icon = t_inhand_state
+		O.icon_state = t_state
+		O.pixel_x = -1*(check_dimensions.Width() - WORLD_ICON_SIZE)/2
+		O.pixel_y = -1*(check_dimensions.Height() - WORLD_ICON_SIZE)/2
+		O.layer = O.layer
+
+		var/list/offsets = get_item_offset_by_index(index)
+
+		O.pixel_x += offsets["x"]
+		O.pixel_y += offsets["y"]
+
+		O.pixel_x += clothing_offset_x
+		O.pixel_y += clothing_offset_y
+
+		if(I.dynamic_overlay && I.dynamic_overlay["[MANNEQUIN_HAND_LAYER]-[index]"])
+			var/image/dyn_overlay = I.dynamic_overlay["[MANNEQUIN_HAND_LAYER]-[index]"]
+			O.overlays.Add(dyn_overlay)
+
+		overlays.Add(O)
+
+
+/obj/structure/mannequin/proc/update_icon_left_hand()
+	return update_icon_hand(GRASP_LEFT_HAND)
+
+/obj/structure/mannequin/proc/update_icon_right_hand()
+	return update_icon_hand(GRASP_RIGHT_HAND)
+
+
+/obj/structure/mannequin/proc/spin()
+	change_dir(turn(dir, 90))
+
+/obj/structure/mannequin/verb/rotate_mannequin()
+	set name = "Rotate Mannequin"
+	set category = "Object"
+	set src in oview(1)
+
+	if(!usr || !isturf(usr.loc))
+		return
+
+	if(anchored)
+		return
+
+	if(usr.isUnconscious() || usr.restrained())
+		return
+
+	spin()
+
+
+/obj/structure/mannequin/fat
+	name = "fat human marble mannequin"
+	icon_state="mannequin_marble_fat_human"
+	fat = 1
+
+/obj/structure/mannequin/vox
+	name = "vox marble mannequin"
+	icon_state="mannequin_marble_vox"
+	species_type = /datum/species/vox
+
+/obj/structure/mannequin/wood
+	name = "human wooden mannequin"
+	desc = "This should look great in a visual arts workshop."
+	icon_state="mannequin_wooden_human"
+	health = 30
+	maxHealth = 30
+
+/obj/structure/mannequin/wood/Destroy()
+	getFromPool(/obj/item/stack/sheet/wood, loc, 5)//You get half the materials used to make a block back
+	..()
+
+/obj/structure/mannequin/wood/fat
+	name = "fat human wooden mannequin"
+	icon_state="mannequin_wooden_fat_human"
+	fat = 1
+
+/obj/structure/mannequin/wood/vox
+	name = "vox wooden mannequin"
+	icon_state="mannequin_wooden_vox"
+	species_type = /datum/species/vox
+
+
+
+
+
+/////////////////////////////////////////////////////////BLOCKS//////////////////////////////////////////////////////////
+//Use a chisel on those to sculpt them into mannequins.
+
+/obj/structure/block
+	name = "marble block"
+	desc = "Grab your chisel and get to work!"
+	anchored = 0
+	density = 1
+	icon = 'icons/obj/mannequin.dmi'
+	icon_state = "marble"
+	var/time_to_sculpt = 200
+	var/result = "/obj/structure/mannequin"
+	var/no_drop = 0
+
+/obj/structure/block/attackby(var/obj/item/weapon/W,var/mob/user)
+	if(iswrench(W))
```

Comment:
There's already a wrenchAnchor proc on /obj/, so you should use that. Same for the mannequin.